eclipse-cdt-cloud/vscode-memory-inspector
# Add Debug Context Selector

#### What it does

This PR addresses [Issue 67](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/issues/67).
In a system with multiple children such as the `cdt-gdb-amalgamator`, the Memory Inspector needs a way to select the correct context (as the `cdt-gdb-vscode` Memory Browser does).

This PR has a dependency on `cdt-gdb-amalgamator` [PR 20](https://github.com/eclipse-cdt-cloud/cdt-amalgamator/pull/20) which puts the assignment of the Child IDs in the `cdt-gdb-amalgamator` instead of the Memory Inspector and also uses the `readMemory` since the data returned is in the correct format for Memory Inspector.

#### How to test

Run this branch of the `vscode-memory-inspector` with the mainline `cdt-gdb-vscode` and `cdt-gdb-adapter` along with the modified `cdt-gdb-amalgamator` in [PR 20](https://github.com/eclipse-cdt-cloud/cdt-amalgamator/pull/20).

Start the debugger as a Desktop Extension. In the newly launched VSCode, open the cdt-amalgamator/sampleWorkspace folder and place a breakpoint at the first instruction in empty1.c's main and empty2.c's main.  Start debugging with `Amalgamator Example` and you should run to the breakpoints. Open a new Memory Inspector with the Command Palette's `Memory: Show Memory Inspector` to open a new view. The dropdown should list both `proc1` and `proc2` listed in the Call Stack.

Select the appropriate `proc` by clicking on the corresponding main in the call stack. View the variable value in memory by typing:
`&local_in_main` into the `Address` textfield and pressing the `Go` button. Go back to the corresponding C file (empty1.c or empty2.c) and single step. The value of the `Local` variable in the `Variables` view will update. Open the `Memory` tab and the `Data` should have changed as well to match the `Variables` view.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the code of conduct](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/blob/main/CODE_OF_CONDUCT.md)
